src/ui.rs

```diff
@@ -0,0 +1,335 @@
+        "Press X to exit the game",
+        screen_width * TEXT_POS_X,
+        screen_height * TEXT_POS_Y + 15.,
+        TEXT_SIZE,
+        BLACK,
+    );
+    draw_text(
+        "Press R to restart the game with a random seed",
+        screen_width * TEXT_POS_X,
+        screen_height * TEXT_POS_Y + 30.,
+        TEXT_SIZE,
+        BLACK,
+    );
+}
+
+/// Runs the game in playing mode, where time passes and the game is updated.
+fn run_playing_mode(game: &mut Game, generations_passed: &mut i32) {
+    sleep(Duration::from_secs(1));
+    game.update_game();
+
+    show_playing_hud();
+    *generations_passed += 1;
+}
+
+/// Displays the HUD for playing mode, showing instructions for the user.
+fn show_playing_hud() {
+    let screen_width = screen_width();
+    let screen_height = screen_height();
+
+    draw_text(
+        "Conway's Game of Life",
+        screen_width * TITLE_POS_X,
+        screen_height * TITLE_POS_Y,
+        TITLE_SIZE,
+        BLACK,
+    );
+    draw_text(
+        "Press SPACE to pause",
+        screen_width * TEXT_POS_X,
+        screen_height * TEXT_POS_Y,
+        TEXT_SIZE,
+        BLACK,
+    );
+    draw_text(
+        "Press X to exit the game",
+        screen_width * TEXT_POS_X,
+        screen_height * TEXT_POS_Y + 15.,
+        TEXT_SIZE,
+        BLACK,
+    );
+    draw_text(
+        "Press R to restart the game with a random seed",
+        screen_width * TEXT_POS_X,
+        screen_height * TEXT_POS_Y + 30.,
+        TEXT_SIZE,
+        BLACK,
+    );
+}
+
+/// Handles users commands.
+fn handle_commands(
+    game: &mut Game,
+    game_running: &mut bool,
+    game_paused: &mut bool,
+    edit_mode: &mut bool,
+) {
+    if is_key_pressed(KeyCode::X) {
+        *game_running = !*game_running;
+    }
+
+    if is_key_pressed(KeyCode::Space) {
+        *game_paused = !*game_paused;
+    }
+
+    if is_key_pressed(KeyCode::E) {
+        if *game_paused {
+            *edit_mode = !*edit_mode;
+        } else {
+            *game_paused = !*game_paused;
+            *edit_mode = !*edit_mode;
+        }
+    }
+
+    if is_key_pressed(KeyCode::R) {
+        restart_game_with_random_seed(game);
+    }
+}
+
+/// Restarts the game with a random seed. The amount of cells and their positions are randomly generated.
+fn restart_game_with_random_seed(game: &mut Game) {
+    let mut rng = ::rand::thread_rng();
+
+    let num_cells = rng.gen_range(1..=(SQUARES * SQUARES) as usize);
+
+    let mut seed = Vec::with_capacity(num_cells);
+
+    for _ in 0..num_cells {
+        let x = rng.gen_range(0..SQUARES) as i32;
+        let y = rng.gen_range(0..SQUARES) as i32;
+        seed.push((x, y));
+    }
+
+    *game = Game::new_with_seed(seed);
+}
+
+/// Runs the user interface of the Game of Life.
+///
+/// Creates a new instance of the game with a default seed
+/// and manages the game loop. Handles user input and updates the game state
+/// based on user interactions.
+///
+pub async fn run_ui() {
```

Comment:
This function could receive the `game: Game`, so the user can easily decide the initial seed by passing it

Reply:
Done!